crates/pairwise_nostd_internal/src/twopoint/idx_3d_offset.rs

```diff
@@ -0,0 +1,725 @@
+/// in the sequences. In more detail:
+/// - instances are initialized to hold every [`Idx3DOffset`] that can
+///   describe every pair of meausrements in either:
+///   1. a single [`CartesianBlock`] instance (see [`Self::new_auto`])
+///   2. from a pair of [`CartesianBlock`]s (see [`Self::new_cross`])
+///
+/// - this type implements the methods of a sequence. The number of contained
+///   [`Idx3DOffset`] instances in a sequence is provided by [`Self::len`],
+///   and you can access the `i`th entry with [`Self::get`]
+///
+/// - this is a lazy sequence in the sense that the [`Idx3DOffset`] instances
+///   are computed when they are accessed (i.e. there isn't an underlying
+///   array containing every [`Idx3DOffset`] instance). This is similar to
+///   "range" types in the Python and Julia programming languages (**note:
+///   Rust's "range" types are very different**)
+///
+/// # Developer Docs: Internal representation
+///
+/// <div class="warning"> These details can/will change @ any time. </div>
+///
+/// This is a little complicated. So we are going to go into gory detail. We
+/// will primarily frame this discussion in terms of the case where we want
+/// pairs drawn from 2 separate [`CartesianBlock`]s (we explain how we handle
+/// generalize to the case with a single [`CartesianBlock`] at the very end).
+///
+/// ## Pairs from separate [`CartesianBlock`]s
+/// In order to implement a lazy sequence of the members of the family of
+/// [`Idx3DOffset`] instances that describes every unique measurement pair
+/// drawn from drawn from 2 separate [`CartesianBlock`] instances, block_a
+/// and block_b, we basically need 2 things:
+/// 1. the number of "family members," `n_family_members`
+/// 2. logic to map every non-negative integer `i` that is less than
+///    `n_family_members` to a distinct "family member"
+///
+/// With that in mind, let's think about how we might enumerate these family
+/// members. For 2 [`CartesianBlock`] instances, we are going to assert that
+/// you can construct a 3D array that holds the every unique [`Idx3DOffset`]
+/// instance that describes a unique measurement pair. Lets call this
+/// ``offset_arr3D``. If this isn't intuitive, its instructive to:
+/// - the measurements at the 8 corners of block_a.
+/// - For each of these corner measurements, consider all pairs of containing
+///   the corner measurement. It should be straight-forward to see that the
+///   set of [`Idx3DOffset`] instances for each pair can be organized into a
+///   3D array. The length along the ith axis comes from either the shape
+///   of block_a or block_b. We'll call each of these arrays of
+///   [`Idx3DOffset`] instances a `offset_subarr3D`.
+/// - You can imagine stitching each of these 8 `offset_subarr3D`s together
+///   into a single giant 3D array. (In the process you would you remove all
+///   duplicate instances of [`Idx3DOffset`] instances)
```

Comment:
I think I would drop this part of the explanation because (unless I'm confused), it's not how the code works.  It was kind of a red herring for me, and I think the following explanations of the size and 3D array of offsets are very clear.